### DISCUSSION TITLE
Is it possible to get a set of docs to describle the manual step by step postal deployment ?

### DISCUSSION THREAD

**Discussion by ghevge** [postalserver/postal#2572] [Feature suggestions]
The current postal installation documentation gravitates around the set of scripts you guys build for docker deployment.
These screaps are not always usefull particularly if you are trying to deploy in  k8s or docker-compose .

Is it possible to provide instead a step by step manual installation procedure document for postal deployment, so that anyone who would like to do any other automation can do it without having to reverse engeneer your scripts?

Thanks

**Comment by willpower232** [comment]
I think someone mentioned a helm chart once but can't remember. The official way uses docker-compose and includes built in things for executing the setup commands and confirming the overall status of the Postal installation so definitely the best answer for most.

Postal requires direct access to ports for SMTP purposes which instantly adds a significant layer of complication that each person will handle differently and can also be quite problematic to set up with kubernetes which means you would have to be (or become) quite knowledgeable about it.

If you're knowledgeable about setting things up in kubernetes then I'd hope you were knowledgeable enough to have a look at the docker-compose.yml file which is part of the official installation and translate that into whatever combination of pods you desired but offering support for all the many ways Postal can be hosted would be quite a toll and challenging to support.

**Comment by ghevge** [comment]
Yes I've already looked at the docker-compose.yaml and some other components in the repo, but my problem is that you guys are using your own scripts to auto generate secret keys also to initialize and create users for postal, etc. These steps are not detailed on how I can generate them  without using your scripts. That is why I was saying that a manual step by step procedure will be more helpful than providing a bunch of scripts to do all these just for docker. Once I will have this info I can automate any deployment I want and won't be limited to docker deployments.
